framework/harness/harness.go

```diff
@@ -34,6 +34,7 @@ func NewTestHarness(
 	testServiceBaseURL string,
 	testHarnessExternalHostname string,
 	testHarnessPort int,
+	https bool,
```

Comment:
How to tell the harness to use HTTPS. It could go like this:
1. Query the service under test
2. If it has `tls`, then start the HTTPS server 
3. Otherwise start the HTTP server

But this means we can't test both TLS and no TLS, it's just one or the other. 

Could solve this like:

* Run only a subset of tests that are representative. Then run them 2x in both configs.
* Have the test harness run the whole suite twice automatically
* Don't worry about testing only https. That is after all the production configuration. Although, sometimes people talk to Relays without certificates (majority of the time maybe.)
* ??